src/NUnitFramework/framework/Internal/Filters/NotFilter.cs

```diff
@@ -76,10 +76,9 @@ public override bool Match( ITest test )
         /// </summary>
         /// <param name="test">The test to which the filter is applied</param>
         /// <returns>True if the test matches the filter explicityly, otherwise false</returns>
-        // TODO: Not yet clear how explicit match should work for NotFilter
         public override bool IsExplicitMatch(ITest test)
         {
-            return !BaseFilter.IsExplicitMatch(test);
+            return !BaseFilter.Match(test);
```

Comment:
Need a test or tests for this. In fairness, we needed them before this, but if we're going to fix it then let's have some tests that show when a NotFilter match is considered Explicit and when it isn't.